common/src/main/java/org/opensearch/ml/common/transport/memorycontainer/memory/MLCreateEventInput.java

```diff
@@ -33,76 +28,96 @@
 import org.opensearch.core.xcontent.ToXContentObject;
 import org.opensearch.core.xcontent.XContentBuilder;
 import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.ml.common.memorycontainer.ShortTermMemoryType;
+import org.opensearch.ml.common.utils.StringUtils;
 
 import lombok.Builder;
 import lombok.Getter;
 import lombok.Setter;
-import org.opensearch.ml.common.memorycontainer.ShortTermMemoryType;
-import org.opensearch.ml.common.utils.StringUtils;
 
 /**
- * Input data for adding memory to a memory container
+ * Input payload for creating short-term events in a memory container
  */
 @Getter
 @Setter
 @Builder
-public class MLAddMemoriesInput implements ToXContentObject, Writeable {
+public class MLCreateEventInput implements ToXContentObject, Writeable {
 
     // Required fields
     private String memoryContainerId;
-    private ShortTermMemoryType memoryType;
+    private ShortTermMemoryType memoryType;  // Auto-detected from fields
     private List<MessageInput> messages;
-    private String binaryData;
-    private Map<String, Object> structuredData;
+    private Map<String, Object> data;
 
     // Optional fields
     private Map<String, String> namespace;
     private boolean infer;
     private Map<String, String> metadata;
-    private Map<String, String> tags;
 
-    public MLAddMemoriesInput(
+    public MLCreateEventInput(
         String memoryContainerId,
         ShortTermMemoryType memoryType,
         List<MessageInput> messages,
-        String binaryData,
-        Map<String, Object> structuredData,
+        Map<String, Object> data,
         Map<String, String> namespace,
         boolean infer,
-        Map<String, String> metadata,
-        Map<String, String> tags
+        Map<String, String> metadata
     ) {
         // MAX_MESSAGES_PER_REQUEST limit removed for performance testing
 
         this.memoryContainerId = memoryContainerId;
-        this.memoryType = memoryType == null ? ShortTermMemoryType.CONVERSATION : memoryType;
         this.messages = messages;
-        this.binaryData = binaryData;
-        this.structuredData = structuredData;
+        this.data = data;
         this.namespace = namespace;
-        this.infer = infer; // default infer is false
         this.metadata = metadata;
-        this.tags = tags;
+
+        // Auto-detect memory type and set infer defaults
+        if (messages != null && !messages.isEmpty()) {
+            this.memoryType = ShortTermMemoryType.CONVERSATIONAL;
+            this.infer = infer; // User can override
+        } else if (data != null) {
+            this.memoryType = ShortTermMemoryType.DATA;
+            this.infer = false; // Always false for data
+        } else {
+            this.memoryType = memoryType; // Use provided type if neither messages nor data
+        }
```

Comment:
We should always set memory type as user provided value, no matter the messages/data value is null or not.

```suggestion
        if (memoryType != null) {
            this.memoryType = memoryType;
        } else if (messages != null && !messages.isEmpty()) {
            this.memoryType = ShortTermMemoryType.CONVERSATIONAL;
            this.infer = infer; // User can override
        } else if (data != null) {
            this.memoryType = ShortTermMemoryType.DATA;
            this.infer = false; // Always false for data
        }
```

Reply:
I agreed that user might have requirements in pre-load messages as data. But I still insist we don't need infer and type together.